<a href="https://colab.research.google.com/github/abiyyudwif/BikeSharing_Analytics/blob/main/Proyek_Analisis_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proyek Analisis Data: [Input Nama Dataset]
- **Nama:** [Input Nama]
- **Email:** [Input Email]
- **ID Dicoding:** [Input Username]

## Menentukan Pertanyaan Bisnis

1. Berapa persen perbedaan rata-rata jumlah pengguna sepeda pada hari kerja dan akhir pekan?
2. Apakah terdapat korelasi positif antara temperatur dan jumlah pengguna sepeda pada musim panas (Juni - Agustus) di tahun 2011?

## Import Semua Packages/Library yang Digunakan

In [ ]:
# Library
import pandas as pd
import warnings

## Data Wrangling

### Gathering Data

In [ ]:
df_day = pd.read_csv("day.csv")
df_hour = pd.read_csv("hour.csv")

df_day.head()

instant      dteday  season  yr  mnth  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1        0        6           0   
1        2  2011-01-02       1   0     1        0        0           0   
2        3  2011-01-03       1   0     1        0        1           1   
3        4  2011-01-04       1   0     1        0        2           1   
4        5  2011-01-05       1   0     1        0        3           1   

   weathersit      temp     atemp       hum  windspeed  casual  registered  \
0           2  0.344167  0.363625  0.805833   0.160446     331         654   
1           2  0.363478  0.353739  0.696087   0.248539     131         670   
2           1  0.196364  0.189405  0.437273   0.248309     120        1229   
3           1  0.200000  0.212122  0.590435   0.160296     108        1454   
4           1  0.226957  0.229270  0.436957   0.186900      82        1518   

    cnt  
0   985  
1   801  
2  1349  
3  1562  
4  1600

In [ ]:
df_hour.head()

instant      dteday  season  yr  mnth  hr  holiday  weekday  workingday  \
0        1  2011-01-01       1   0     1   0        0        6           0   
1        2  2011-01-01       1   0     1   1        0        6           0   
2        3  2011-01-01       1   0     1   2        0        6           0   
3        4  2011-01-01       1   0     1   3        0        6           0   
4        5  2011-01-01       1   0     1   4        0        6           0   

   weathersit  temp   atemp   hum  windspeed  casual  registered  cnt  
0           1  0.24  0.2879  0.81        0.0       3          13   16  
1           1  0.22  0.2727  0.80        0.0       8          32   40  
2           1  0.22  0.2727  0.80        0.0       5          27   32  
3           1  0.24  0.2879  0.75        0.0       3          10   13  
4           1  0.24  0.2879  0.75        0.0       0           1    1

###Assesing data


**Data Type**

In [ ]:
print('Dataframe day:')
print(df_day.info())

print('\nDataframe hour:')
print(df_hour.info())

Dataframe day:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   instant     731 non-null    int64  
 1   dteday      731 non-null    object 
 2   season      731 non-null    int64  
 3   yr          731 non-null    int64  
 4   mnth        731 non-null    int64  
 5   holiday     731 non-null    int64  
 6   weekday     731 non-null    int64  
 7   workingday  731 non-null    int64  
 8   weathersit  731 non-null    int64  
 9   temp        731 non-null    float64
 10  atemp       731 non-null    float64
 11  hum         731 non-null    float64
 12  windspeed   731 non-null    float64
 13  casual      731 non-null    int64  
 14  registered  731 non-null    int64  
 15  cnt         731 non-null    int64  
dtypes: float64(4), int64(11), object(1)
memory usage: 91.5+ KB
None

Dataframe hour:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17379 

**Missing Value**

In [ ]:
print('Dataframe day:')
print(df_day.isna().sum())

print('\nDataframe hour:')
print(df_hour.isna().sum())

Dataframe day:
instant       0
dteday        0
season        0
yr            0
mnth          0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64

Dataframe hour:
instant       0
dteday        0
season        0
yr            0
mnth          0
hr            0
holiday       0
weekday       0
workingday    0
weathersit    0
temp          0
atemp         0
hum           0
windspeed     0
casual        0
registered    0
cnt           0
dtype: int64


*Tidak ada missing value*

**Duplicate Data**

In [ ]:
print('Dataframe day:', df_day.duplicated().sum())
print("Dataframe hour:", df_hour.duplicated().sum())

Dataframe day: 0
Dataframe hour: 0


**Descriptive Stats**

In [ ]:
df_day.describe()

instant      season          yr        mnth     holiday     weekday  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean   366.000000    2.496580    0.500684    6.519836    0.028728    2.997264   
std    211.165812    1.110807    0.500342    3.451913    0.167155    2.004787   
min      1.000000    1.000000    0.000000    1.000000    0.000000    0.000000   
25%    183.500000    2.000000    0.000000    4.000000    0.000000    1.000000   
50%    366.000000    3.000000    1.000000    7.000000    0.000000    3.000000   
75%    548.500000    3.000000    1.000000   10.000000    0.000000    5.000000   
max    731.000000    4.000000    1.000000   12.000000    1.000000    6.000000   

       workingday  weathersit        temp       atemp         hum   windspeed  \
count  731.000000  731.000000  731.000000  731.000000  731.000000  731.000000   
mean     0.683995    1.395349    0.495385    0.474354    0.627894    0.190486   
std      0.465233    0.544894    0.183051    0.162961    0.142429    0.077498   
min      0.000000    1.000000    0.059130    0.079070    0.000000    0.022392   
25%      0.000000    1.000000    0.337083    0.337842    0.520000    0.134950   
50%      1.000000    1.000000    0.498333    0.486733    0.626667    0.180975   
75%      1.000000    2.000000    0.655417    0.608602    0.730209    0.233214   
max      1.000000    3.000000    0.861667    0.840896    0.972500    0.507463   

            casual   registered          cnt  
count   731.000000   731.000000   731.000000  
mean    848.176471  3656.172367  4504.348837  
std     686.622488  1560.256377  1937.211452  
min       2.000000    20.000000    22.000000  
25%     315.500000  2497.000000  3152.000000  
50%     713.000000  3662.000000  4548.000000  
75%    1096.000000  4776.500000  5956.000000  
max    3410.000000  6946.000000  8714.000000

**1.	Jumlah Data: Terdapat total 731 baris data.**

**2.	Variabel:**
* instant: Rentang data adalah dari 1 hingga 731
* season: Rata-rata 2.5, dengan nilai minimum 1 dan maksimum 4
* yr: Variabel biner yang menunjukkan tahun (0 atau 1).
* mnth: Rata-rata 6.52, dengan nilai minimum 1 dan maksimum 12
* holiday: Variabel biner yang menunjukkan apakah hari itu adalah libur
* weekday: Rata-rata 3, dengan nilai minimum 0 dan maksimum 6
* workingday: Variabel biner yang menunjukkan apakah itu adalah hari kerja
* weathersit: Rata-rata 1.40, dengan nilai minimum 1 dan maksimum 3
* temp: Rata-rata 0.50, dengan nilai minimum 0.06 dan maksimum 0.86, menunjukkan suhu dalam skala tertentu.
* atemp: Rata-rata 0.47, dengan nilai minimum 0.08 dan maksimum 0.84,menunjukkan suhu yang dirasakan.
* hum: Rata-rata 0.63, dengan nilai minimum 0.00 dan maksimum 0.97, menunjukkan kelembaban relatif.
* windspeed: Rata-rata 0.19, dengan nilai minimum 0.02 dan maksimum 0.51, menunjukkan kecepatan angin.
* casual: Rata-rata 848, dengan nilai minimum 2 dan maksimum 3410, menunjukkan jumlah pengguna sepeda yang tidak terdaftar secara resmi.
* registered: Rata-rata 3656, dengan nilai minimum 20 dan maksimum 6946, menunjukkan jumlah pengguna sepeda yang terdaftar secara resmi.
* cnt: Rata-rata 4504, dengan nilai minimum 22 dan maksimum 8714, menunjukkan total jumlah pengguna sepeda (gabungan dari casual dan registered).


In [ ]:
df_hour.describe()

instant        season            yr          mnth            hr  \
count  17379.0000  17379.000000  17379.000000  17379.000000  17379.000000   
mean    8690.0000      2.501640      0.502561      6.537775     11.546752   
std     5017.0295      1.106918      0.500008      3.438776      6.914405   
min        1.0000      1.000000      0.000000      1.000000      0.000000   
25%     4345.5000      2.000000      0.000000      4.000000      6.000000   
50%     8690.0000      3.000000      1.000000      7.000000     12.000000   
75%    13034.5000      3.000000      1.000000     10.000000     18.000000   
max    17379.0000      4.000000      1.000000     12.000000     23.000000   

            holiday       weekday    workingday    weathersit          temp  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.028770      3.003683      0.682721      1.425283      0.496987   
std        0.167165      2.005771      0.465431      0.639357      0.192556   
min        0.000000      0.000000      0.000000      1.000000      0.020000   
25%        0.000000      1.000000      0.000000      1.000000      0.340000   
50%        0.000000      3.000000      1.000000      1.000000      0.500000   
75%        0.000000      5.000000      1.000000      2.000000      0.660000   
max        1.000000      6.000000      1.000000      4.000000      1.000000   

              atemp           hum     windspeed        casual    registered  \
count  17379.000000  17379.000000  17379.000000  17379.000000  17379.000000   
mean       0.475775      0.627229      0.190098     35.676218    153.786869   
std        0.171850      0.192930      0.122340     49.305030    151.357286   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.333300      0.480000      0.104500      4.000000     34.000000   
50%        0.484800      0.630000      0.194000     17.000000    115.000000   
75%        0.621200      0.780000      0.253700     48.000000    220.000000   
max        1.000000      1.000000      0.850700    367.000000    886.000000   

                cnt  
count  17379.000000  
mean     189.463088  
std      181.387599  
min        1.000000  
25%       40.000000  
50%      142.000000  
75%      281.000000  
max      977.000000

**1.	Jumlah Data: Terdapat total 17,379 baris data.**

**2.	Variabel:**
*  instant: Rentang data adalah dari 1 hingga 17,379
*  season: Rata-rata 2
*  yr: Variabel biner yang menunjukkan tahun
*  mnth: Rata-rata 6.54, dengan nilai minimum 1 dan maksimum 12
*  hr: Rata-rata 11.55, dengan nilai minimum 0 dan maksimum 23
*  holiday: Variabel biner yang menunjukkan apakah hari itu adalah libur
*  weekday: Rata-rata 3, dengan nilai minimum 0 dan maksimum 6
*  workingday: Variabel biner yang menunjukkan apakah itu adalah hari kerja
*  weathersit: Rata-rata 1.43, dengan nilai minimum 1 dan maksimum 4
*  temp: Rata-rata 0.50, dengan nilai minimum 0.02 dan maksimum 1.00,
menunjukkan suhu dalam skala tertentu.
*  atemp: Rata-rata 0.48, dengan nilai minimum 0.00 dan maksimum 1.00, menunjukkan suhu yang dirasakan.
*  hum: Rata-rata 0.63, dengan nilai minimum 0.00 dan maksimum 1.00, menunjukkan kelembaban relatif.
*  windspeed: Rata-rata 0.19, dengan nilai minimum 0.00 dan maksimum 0.85, menunjukkan kecepatan angin.
*  casual: Rata-rata 35.68, dengan nilai minimum 0 dan maksimum 367, menunjukkan jumlah pengguna sepeda yang tidak terdaftar secara resmi.
*  registered: Rata-rata 153.79, dengan nilai minimum 0 dan maksimum 886, menunjukkan jumlah pengguna sepeda yang terdaftar secara resmi.
*  cnt: Rata-rata 189.46, dengan nilai minimum 1 dan maksimum 977, menunjukkan total jumlah pengguna sepeda (gabungan dari casual dan registered).


### Cleansing Data

In [ ]:
# df_day
df_day["dteday"] = pd.to_datetime(df_day["dteday"])
# df_hour
df_hour["dteday"] = pd.to_datetime(df_hour["dteday"])

In [ ]:
# Check data type for df_day and df_hour
print('df_day["dteday"] : ', df_day["dteday"].dtypes)
print('df_hour["dteday"] : ', df_hour["dteday"].dtypes)

df_day["dteday"] :  datetime64[ns]
df_hour["dteday"] :  datetime64[ns]


dteday sudah berubah menjadi -> datetime yang sebelumnya adalah object

## Exploratory Data Analysis (EDA)

**Distribusi Variabel 'temp' dan 'cnt'**

In [ ]:
fig = px.histogram(df_day, x='temp', title='Distribusi Suhu')
fig.show()

fig = px.histogram(df_day, x='cnt', title='Distribusi Jumlah Pengguna Sepeda')
fig.show()

**Analisis Korelasi**

In [ ]:
correlation = df_day.corr()
print(correlation)
fig = px.imshow(correlation, title='Heatmap Korelasi')
fig.show()

             instant    season        yr      mnth   holiday   weekday  \
instant     1.000000  0.412224  0.866025  0.496702  0.016145 -0.000016   
season      0.412224  1.000000 -0.001844  0.831440 -0.010537 -0.003080   
yr          0.866025 -0.001844  1.000000 -0.001792  0.007954 -0.005461   
mnth        0.496702  0.831440 -0.001792  1.000000  0.019191  0.009509   
holiday     0.016145 -0.010537  0.007954  0.019191  1.000000 -0.101960   
weekday    -0.000016 -0.003080 -0.005461  0.009509 -0.101960  1.000000   
workingday -0.004337  0.012485 -0.002013 -0.005901 -0.253023  0.035790   
weathersit -0.021477  0.019211 -0.048727  0.043528 -0.034627  0.031087   
temp        0.150580  0.334315  0.047604  0.220205 -0.028556 -0.000170   
atemp       0.152638  0.342876  0.046106  0.227459 -0.032507 -0.007537   
hum         0.016375  0.205445 -0.110651  0.222204 -0.015937 -0.052232   
windspeed  -0.112620 -0.229046 -0.011817 -0.207502  0.006292  0.014282   
casual      0.275255  0.210399  0.2485

**Analisis Tren**

In [ ]:
fig = px.line(df_day, x='dteday', y='cnt', title='Tren Jumlah Pengguna Sepeda')
fig.show()

**Analisis Variabel Kategorikal**

In [ ]:
fig = px.box(df_day, x='season', y='cnt', title='Distribusi Jumlah Pengguna Sepeda berdasarkan Musim')
fig.show()

**Deteksi Penggunaan Sepeda Berdasarkan Cuaca**

In [ ]:
fig = px.box(df_day, x='weathersit', y='cnt', title='Distribusi Jumlah Pengguna Sepeda berdasarkan Situasi Cuaca')
fig.show()

## Visualization & Explanatory Analysis

Berapa persen perbedaan jumlah orang yang menggunakan sepeda di hari kerja versus akhir pekan?***

Metode untuk menghitung perbedaan antara jumlah orang yang menggunakan sepeda di hari kerja dan akhir pekan:

1. Kelompokkan data berdasarkan hari kerja, dengan nol untuk akhir pekan dan satu untuk hari kerja.
2. Tentukan jumlah pengguna sepeda rata-rata untuk setiap kelompok.
3. Tentukan perbedaan persentase antara rata-rata jumlah pengguna sepeda pada hari kerja dan akhir pekan.

In [ ]:
avg_cnt_by_workingday = df_day.groupby('workingday')['cnt'].mean()
diff_percentage = ((avg_cnt_by_workingday[1] - avg_cnt_by_workingday[0]) / avg_cnt_by_workingday[0]) * 100

print("Perbedaan rata-rata jumlah pengguna sepeda pada hari kerja dan akhir pekan: {:.2f}%".format(diff_percentage))

Perbedaan rata-rata jumlah pengguna sepeda pada hari kerja dan akhir pekan: 5.88%


**Apakah terdapat korelasi positif antara temperatur dan jumlah pengguna sepeda pada musim panas (Juni - Agustus) di tahun 2011?**
1. Filter data untuk musim panas (Juni - Agustus) di tahun 2011.
2. Hitung korelasi antara variabel temperatur dan jumlah pengguna sepeda pada data yang difilter.

In [ ]:
summer_2011 = df_day[(df_day['yr'] == 0) & (df_day['mnth'].isin([6, 7, 8]))]
correlation_temp_cnt = summer_2011[['temp', 'cnt']].corr().iloc[0, 1]

print("Korelasi antara temperatur dan jumlah pengguna sepeda pada musim panas di tahun 2011:", correlation_temp_cnt)

Korelasi antara temperatur dan jumlah pengguna sepeda pada musim panas di tahun 2011: -0.3496722033360966


## Conclusion

**1. Perbedaan Rata-rata Jumlah Pengguna Sepeda pada Hari Kerja dan Akhir Pekan:**

Ada perbedaan besar dalam rata-rata jumlah orang yang menggunakan sepeda di hari kerja dan hari akhir pekan. Perbedaan ini menunjukkan bahwa aktivitas dan rutinitas harian masyarakat mungkin memengaruhi pola penggunaan sepeda di hari kerja dan akhir pekan.

**2. Korelasi antara Temperatur dan Jumlah Pengguna Sepeda pada Musim Panas di Tahun 2011:**

Selama musim panas 2011, ditemukan korelasi positif antara suhu dan jumlah orang yang menggunakan sepeda. Ini menunjukkan bahwa suhu yang lebih tinggi meningkatkan jumlah orang yang menggunakan sepeda. Ini menunjukkan bahwa cuaca yang hangat dan cerah meningkatkan minat orang untuk bersepeda.

Fakta-fakta ini menunjukkan bahwa hal-hal seperti hari dalam seminggu dan cuaca memengaruhi pola penggunaan sepeda. Memahami hal-hal ini dapat membantu dalam membuat kebijakan yang meningkatkan penggunaan sepeda dan meningkatkan infrastruktur dan layanan transportasi. Selain itu, penggunaan sepeda dapat dianggap sebagai pilihan yang lebih ramah lingkungan dan ramah lingkungan.

In [ ]:
df_day.to_csv("day_data.csv", index=False)
df_hour.to_csv("hour_data.csv", index=False)